cmd/osm-bootstrap/osm-bootstrap.go

```diff
@@ -260,19 +275,74 @@ func applyOrUpdateCRDs(crdClient *apiclient.ApiextensionsV1Client) {
-
-			crdExisting.Labels[constants.ReconcileLabel] = strconv.FormatBool(enableReconciler)
-			crdExisting.Spec = crd.Spec
-			crdExisting.Spec.Conversion = &apiv1.CustomResourceConversion{
-				Strategy: apiv1.NoneConverter,
-			}
-			if _, err = crdClient.CustomResourceDefinitions().Update(context.Background(), crdExisting, metav1.UpdateOptions{}); err != nil {
-				log.Fatal().Err(err).Msgf("Error updating conversion webhook configuration for crd : %s", crd.Name)
+			// Save the CRDs name and initialize patchConfirmation to signal that ensureCRDConversionChange should handle the patching
+			originalCRDs = append(originalCRDs, crd)
+			if patchConfirmation == nil {
+				patchConfirmation = make(chan struct{})
 			}
-			log.Info().Msgf("successfully set conversion webhook configuration for crd : %s to \"None\"", crd.Name)
 		}
 	}
+
+	if patchConfirmation != nil {
+		go ensureCRDConversionChange(originalCRDs, crdClient, patchConfirmation)
+	}
+	return patchConfirmation
+}
+
+// ensureCRDConversionChange ensures that the CRD conversion strategy is set to None
+// by blocking until all CRDs are updated to have a conversion strategy of None
+// 3 times over a 15 second period.
+func ensureCRDConversionChange(originalCRDs []*apiv1.CustomResourceDefinition, crdClient *apiclient.ApiextensionsV1Client, patchConfirmation chan<- struct{}) {
+	for counter, totalAttempts := 0, 0; counter < 3; totalAttempts++ {
+		if checkCRDConversionChange(originalCRDs, crdClient, totalAttempts) {
+			counter++
+		} else {
+			counter = 0 // reset the counter if we see a change in the CRD conversion strategy
+		}
+		time.Sleep(500 * time.Millisecond)
+	}
+	close(patchConfirmation)
+}
+
+func checkCRDConversionChange(originalCRDs []*apiv1.CustomResourceDefinition, crdClient *apiclient.ApiextensionsV1Client, attempts int) bool {
+	success := true
+	for _, originalCRD := range originalCRDs {
+		if originalCRD == nil {
+			log.Error().Msg("nil crd detected")
+			continue
+		}
+		clusterCRD, err := crdClient.CustomResourceDefinitions().Get(context.Background(), originalCRD.Name, metav1.GetOptions{})
+		if err != nil {
+			// We just error here sicne the calling function will retry
+			log.Error().Err(err).Msgf("error getting CRD %s", originalCRD.Name)
+			return false
+		}
+		if clusterCRD.Spec.Conversion != nil && clusterCRD.Spec.Conversion.Strategy != apiv1.NoneConverter {
+			log.Info().Msgf("Attempt #%d: CRD %s has conversion strategy %s, expected %s. Patching...", attempts, clusterCRD.Name, clusterCRD.Spec.Conversion.Strategy, "None")
+			success = false
+			patchCRDConversion(originalCRD, clusterCRD, crdClient)
+		} else {
+			log.Info().Msgf("Attempt #%d for CRD %s has succeeded!", attempts, clusterCRD.Name)
+
+		}
+	}
+	return success
+}
+
+func patchCRDConversion(originalCRD, clusterCRD *apiv1.CustomResourceDefinition, crdClient *apiclient.ApiextensionsV1Client) {
```

Comment:
I may be missing something but I'm not sure I see any mechanism that prevents the old bootstrap pod from reconciling a patched CRD

Reply:
The idea behind this PR was that ~nothing could prevent the old bootstrap pod from reconciling since we're on an upgrade path. I'm just noticing that the reconciler informer has a label selector in the lister, _not_ the event handler, so this fix becomes much simpler. Rewriting